var/spack/repos/builtin/packages/py-packaging/package.py

```diff
@@ -20,11 +21,13 @@ class PyPackaging(PythonPackage):
     version('17.1', sha256='f019b770dd64e585a99714f1fd5e01c7a8f11b45635aa953fd41c689a657375b')
     version('16.8', sha256='5d50835fdf0a7edf0b55e311b7c887786504efea1177abd7e69329a8e5ea619e')
 
-    depends_on('python@3.4:', when='@21:', type=('build', 'run'))
-    depends_on('python@2.7:2.8,3.4:', type=('build', 'run'))
+    depends_on('python@3.6:', when='@21:', type=('build', 'run'))
+    depends_on('python@2.7:2,3.4:', type=('build', 'run'))
     depends_on('py-setuptools@40.8.0:', when='@20.8:', type='build')
     depends_on('py-setuptools', type='build')
 
+    depends_on('py-pyparsing@2.0.2:3.0.4,3.0.6:', when='@21.3:', type=('build', 'run'))
+    depends_on('py-pyparsing@2.0.2:2', when='@21.1:', type=('build', 'run'))
```

Comment:
```suggestion
    depends_on('py-pyparsing@2.0.2:2', when='@21.1:21.2', type=('build', 'run'))
```
Otherwise this condition will also apply to 21.3+